pkg/validate/validate.go

```diff
@@ -115,6 +115,17 @@ func ClusterName1035(v string) error {
 	return ClusterName(v)
 }
 
+// GCPClusterName checks if the provided cluster name has words similar to the word 'google'
+// since resources with that name are not allowed in GCP.
+func GCPClusterName(v string) error {
+	reStartsWith := regexp.MustCompile("^goog")
+	reContains := regexp.MustCompile(".*google.*")
```

Comment:
You don't need the `.*` parts. Also, we should check for "0" in place of "o", too.
```suggestion
	reContains := regexp.MustCompile("g[o0]{2}gle")
```